sw/device/lib/dif/dif_clkmgr.h

```diff
@@ -0,0 +1,200 @@
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_clkmgr_result_t dif_clkmgr_init(dif_clkmgr_params_t params,
+                                    dif_clkmgr_t *handle);
+
+/**
+ * Check if a Gateable Clock is Enabled or Disabled.
+ *
+ * @param handle Clock Manager Handle.
+ * @param clock Gateable Clock ID.
+ * @param[out] is_enabled whether the clock is enabled or not.
+ * @returns The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_clkmgr_result_t dif_clkmgr_gateable_clock_is_enabled(
+    const dif_clkmgr_t *handle, dif_clkmgr_gateable_clock_t clock,
+    bool *is_enabled);
+
+/**
+ * Enable or Disable a Gateable Clock.
+ *
+ * @param handle Clock Manager Handle.
+ * @param clock Gateable Clock ID.
+ * @param new_state whether to enable or disable the clock.
+ * @returns The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_clkmgr_result_t dif_clkmgr_gateable_clock_toggle(
+    const dif_clkmgr_t *handle, dif_clkmgr_gateable_clock_t clock,
+    dif_clkmgr_toggle_t new_state);
+
+/**
+ * Check if a Hintable Clock is Enabled or Disabled.
+ *
+ * Hintable clocks are not under full software control, but this operation
+ * accurately reflects the state of the current clock, regardless of any
+ * requests made by software.
+ *
+ * @param handle Clock Manager Handle.
+ * @param clock Hintable Clock ID.
+ * @param[out] is_enabled whether the clock is enabled or not.
+ * @returns The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_clkmgr_result_t dif_clkmgr_hintable_clock_is_enabled(
+    const dif_clkmgr_t *handle, dif_clkmgr_hintable_clock_t clock,
+    bool *is_enabled);
+
+/**
+ * Enable or Disable a Hintable Clock.
+ *
+ * This only sets a hint that the clock should be enabled or disabled. The clock
+ * manager is in control of whether the clock should actually be enabled or
+ * disabled.
+ *
+ * To read what hint the software has given the hardware, use
+ * #dif_clkmgr_hintable_clock_read_hint. To read whether the clock is currently
+ * enabled or disabled, use #dif_clkmgr_hintable_clock_is_enabled.
+ *
+ * @param handle Clock Manager Handle.
+ * @param clock Hintable Clock ID.
+ * @param new_state whether to enable or disable the clock.
+ * @returns The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_clkmgr_result_t dif_clkmgr_hintable_clock_toggle(
+    const dif_clkmgr_t *handle, dif_clkmgr_hintable_clock_t clock,
+    dif_clkmgr_toggle_t new_state);
+
+/**
+ * Read the current Hint for a Hintable Clock.
+ *
+ * Hintable clocks are not under full software control, but this operation
+ * accurately reflects the state of the current clock, regardless of any
+ * requests made by software.
+ *
+ * @param handle Clock Manager Handle.
+ * @param clock Hintable Clock ID.
+ * @param[out] is_enabled the current software request (hint) for this clock.
+ * @returns The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_clkmgr_result_t dif_clkmgr_hintable_clock_read_hint(
```

Comment:
If this winds up being a pair of functions, do you think `dif_clkmgr_get_hint` and `set_hint` are more succinct?

Reply:
I like @mcy suggestion!

Reply:
`set`/`get` are idiomatic with our other DIFs, so these functions should become `hintable_clock_get_enabled`, `hintable_clock_set_hint`, `hintable_clock_get_hint`.